src/ResourceManager/Network/Commands.Network/VirtualNetwork/Subnet/Delegation/AzureDelegationBase.cs

```diff
@@ -0,0 +1,28 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.Network
+{
+    public class AzureDelegationBase : NetworkBaseCmdlet
+    {
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The name of the delegation")]
+        [ValidateNotNullOrEmpty]
+        public virtual string Name { get; set; }
```

Comment:
you're overwritting the Name parameter in each of the classes - please just remove this parameter here and the override from the other parameters in cmdlet classes.

Reply:
I was following the convention from other cmdlets... not sure why they did it that way. Is it better to stick to the convention or to just remove the base class?

Reply:
I think it would be better to just delete the base class - I think others do it that way because they share a lot of parameters with exactly the same parameter attributes whereas yours don't share the exact same attributes.

Reply:
Fixed.